main/src/main/scala/sbt/internal/CommandExchange.scala

```diff
@@ -167,21 +168,35 @@ private[sbt] final class CommandExchange {
   }
 
   def publishEvent[A: JsonFormat](event: A): Unit = {
+    val broadcastStringMessage = true
     val toDel: ListBuffer[CommandChannel] = ListBuffer.empty
     event match {
       case entry: StringEvent =>
+        val params = toLogMessageParams(entry)
         channels.foreach {
           case c: ConsoleChannel =>
-            if (entry.channelName.isEmpty || entry.channelName == Some(c.name)) {
+            if (broadcastStringMessage) {
               c.publishEvent(event)
+            } else {
+              if (entry.channelName.isEmpty || entry.channelName == Some(c.name)) {
+                c.publishEvent(event)
+              }
             }
           case c: NetworkChannel =>
             try {
-              if (entry.channelName == Some(c.name)) {
-                c.publishEvent(event)
+              // Note that language server's LogMessageParams does not hold the execid,
+              // so this is weaker than the StringMessage. We might want to double-send
+              // in case we have a better client that can utilize the knowledge.
```

Comment:
I think `execId` doesn't make much sense for LSP notifications, the `channelName` is much more important:
- notifications are not necessarily caused by some request, so they don't need `execId` (which in case of ResponseMessages is the request ID)
- it's important that notifications go to the right channel. So if something is triggered by an LSP client, console channel shouldn't get spammed by the log messages and another way round.

Reply:
If there are n events coming out of m commands, you need execId to distinguish them from the other.

Reply:
My point is that LSP doesn't distinguish notifications by exec-id. It does distinguish responses, but the problem here is that I see some `StringEvents` sent to the client with empty exec-ids, because they are sent as responses to a request that client didn't make.